### ISSUE TITLE
[no-confusing-arrow]: new line, and multiple parameters should be allowed

### ISSUE THREAD

**Issue by gblazex** [eslint/eslint#8877]
**Tell us about your environment**

* **ESLint Version:** 4.1.1
* **Node Version:** 8.1.3
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
"no-confusing-arrow": [2, {"allowParens": true}],
```

**What did you do? Please include the actual source code causing the issue.**

```js
// case 1: arrow function body in new line
imageURLToBlob(imageURL, (blob) =>
     blob ? resolve(blob) : reject());

// case 2: arrow function with multiple parameters
var fn = (exists, localURL) => exists ? resolve(localURL) : reject();
```

**What did you expect to happen?**
A new line should make the intended arrow function use case of '=>' obvious.
Or there should be an option to accept this as not confusing.

Multiple parameters again should make an operator '=>' unambiguously treated as an arrow function.

**What actually happened? Please include the actual, raw output from ESLint.**
Both cases report arrow functions can be confused with mathematical operators.

**Comment by gyandeeps** [comment]
I think going this far tells me that its better to just turn off the rule.

**Comment by galambalazs** [comment]
The rule has legitimate use-cases

The example listed on the website is clearly ambiguous (confusing).

```js
// The intent is not clear
var x = a => 1 ? 2 : 3;
// Did the author mean this
var x = a => (1 ? 2 : 3);
// Or this
var x = (a => 1) ? 2 : 3;
```

That's what it's meant to catch.
The cases I mentioned in the first post are crystal clear.

**Comment by platinumazure** [comment]
I would say the multiple argument case is definitely a bug. Parentheses are always required for arrow functions in those cases.

Multi-line support, to me, should be handled as an enhancement request (for a new option).

@gyandeeps Do you think we could at least handle the multiple argument case as a bug? The ECMAScript grammar doesn't allow for removing the parentheses without changing the meaning, and it wouldn't make sense for someone to use `(a, b) >= c ? d : e` in most cases, so I agree with OP that for multiple arguments it's probably not confusing at all. Furthermore, if the `allowParens` option already works for single argument arrow functions, I think it should work for multiple argument arrow functions, even if we choose not to change default behavior for that case.

**Comment by not-an-aardvark** [comment]
I think the `allowParens` option is intended to refer to whether the body of the function is parenthesized, not the parameters.

That said, I've always been a bit unclear on what makes the arrow "confusing".

**Comment by kaicataldo** [comment]
> I think the `allowParens` option is intended to refer to whether the body of the function is parenthesized, not the parameters.

This is correct. I think we're talking about different things here. For the two examples given the rule currently enforces the following:
```js
// case 1: arrow function body in new line
imageURLToBlob(imageURL, (blob) =>
     (blob ? resolve(blob) : reject()));

// case 2: arrow function with multiple parameters
var fn = (exists, localURL) => (exists ? resolve(localURL) : reject());
```

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
